packages/fuel-indexer-lib/src/lib.rs

```diff
@@ -36,6 +36,70 @@ impl ExecutionSource {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum WasmIndexerError {
+    DeserializationError = 1,
+    SerializationError,
+    PutObjectError,
+    UnableToSaveListType,
+    UninitializedMemory,
+    UnableToFetchLogString,
+    KillSwitch,
+    DatabaseError,
+    GeneralError,
+}
+
+impl From<i32> for WasmIndexerError {
+    fn from(value: i32) -> Self {
+        match value {
+            0 => unreachable!("WasmIndexerError index starts at 1"),
+            1 => Self::DeserializationError,
+            2 => Self::SerializationError,
+            3 => Self::PutObjectError,
+            4 => Self::UnableToSaveListType,
+            5 => Self::UninitializedMemory,
+            6 => Self::UnableToFetchLogString,
+            7 => Self::KillSwitch,
+            8 => Self::DatabaseError,
+            _ => Self::GeneralError,
+        }
+    }
+}
+
+impl std::fmt::Display for WasmIndexerError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::SerializationError => {
+                write!(f, "Failed to serialize")
+            }
+            Self::DeserializationError => {
+                write!(f, "Failed to deserialize")
+            }
+            Self::UnableToSaveListType => {
+                write!(f, "Failed to save list")
+            }
+            Self::PutObjectError => {
+                write!(f, "Failed to save object")
+            }
+            Self::UninitializedMemory => {
+                write!(f, "Failed to create MemoryView for indexer")
+            }
+            Self::UnableToFetchLogString => {
+                write!(f, "Failed to fetch log string")
+            }
+            Self::KillSwitch => {
+                write!(f, "Kill switch has been triggered")
```

Comment:
It's not clear what "Kill switch" means in this case.
```suggestion
                write!(f, "Indexer kill switch has been triggered. Indexer will halt.")
```